Merge tag Homebrew/1.1.11

bw-yuhogg opened the pull request:
Due to the magnitude of changes here, IMO we should not merge this for a while.

joshka committed:
Install tap command completions and manpages

Taps can include completion scripts for external commands under
`completions/bash`, `completions/fish`, or `completions/zsh`. `brew tap`
will automatically install these into the correct directories during
install.

reitermarkus committed:
Use `--force-default-config` option for `brew cask style`.

reitermarkus committed:
Convert Formula test to spec.

reitermarkus committed:
Convert `brew install` test to spec.

reitermarkus committed:
Convert `brew uninstall` test to spec.

reitermarkus committed:
Merge pull request #2159 from reitermarkus/spec-cleanup

Convert `brew cleanup` test to spec.

reitermarkus committed:
Merge pull request #2155 from reitermarkus/spec-formula-cmd

Convert `brew formula` test to spec.

MikeMcQuaid committed:
README: add Netlify link.

reitermarkus committed:
Merge pull request #2132 from reitermarkus/spec-options

Convert `brew options` test to spec.

MikeMcQuaid committed:
Merge pull request #1637 from broder/mirror_audit

Updated resource auditing to detect invalid mirrors when using --online

woodruffw committed:
Add William Woodruff, JCount to maintainers lists

joshliu committed:
Change deps_spec expect ordering

reitermarkus committed:
Merge pull request #2147 from reitermarkus/spec-custom-external-command

Convert `brew custom-external-command` test to spec.

reitermarkus committed:
Merge pull request #2135 from reitermarkus/spec-home

Convert `brew home` test to spec.

reitermarkus committed:
Merge pull request #2133 from reitermarkus/spec-test

Convert `brew test` test to spec.

reitermarkus committed:
Merge pull request #2152 from reitermarkus/spec-commands

Convert `brew commands` test to spec.

reitermarkus committed:
Merge pull request #2130 from reitermarkus/spec-fetch

Convert `brew fetch` test to spec.

reitermarkus committed:
Merge pull request #2143 from reitermarkus/spec-leaves

Convert `brew leaves` test to spec.

reitermarkus committed:
Merge pull request #2141 from reitermarkus/spec-outdated

Convert `brew outdated` test to spec.

reitermarkus committed:
Merge pull request #2116 from reitermarkus/spec-list

Convert `brew list` test to spec.

reitermarkus committed:
Merge pull request #2148 from reitermarkus/spec---version

Convert `brew --version` test to spec.

reitermarkus committed:
Merge pull request #2150 from reitermarkus/spec---prefix

Convert `brew --prefix` test to spec.

reitermarkus committed:
Merge pull request #2145 from reitermarkus/spec-tap

Convert `brew tap` test to spec.

reitermarkus committed:
Merge pull request #2136 from reitermarkus/spec-tap-new

Convert `brew tap-new` test to spec.

reitermarkus committed:
Merge pull request #2153 from reitermarkus/spec-pull

Convert `brew pull` test to spec.

reitermarkus committed:
Merge pull request #2138 from reitermarkus/spec-irb

Convert `brew irb` test to spec.

reitermarkus committed:
Merge pull request #2165 from reitermarkus/spec-bottle

Convert `brew bottle` test to spec.

reitermarkus committed:
Merge pull request #2161 from reitermarkus/spec-cask

Convert `brew cask` test to spec.

reitermarkus committed:
Merge pull request #2121 from reitermarkus/spec-services

Convert `brew services` test to spec.

reitermarkus committed:
Merge pull request #2149 from reitermarkus/spec---repository

Convert `brew --repository` test to spec.

reitermarkus committed:
Merge pull request #2129 from reitermarkus/spec-doctor

Convert `brew doctor` test to spec.

reitermarkus committed:
Merge pull request #2162 from reitermarkus/spec---cellar

Convert `brew --cellar` test to spec.

reitermarkus committed:
Merge pull request #2156 from reitermarkus/spec-edit

Convert `brew edit` test to spec.

reitermarkus committed:
Merge pull request #2117 from reitermarkus/spec-pin_unpin

Convert `brew pin` and `brew unpin` test to spec.

reitermarkus committed:
Merge pull request #2139 from reitermarkus/spec-missing

Convert `brew missing` test to spec.

reitermarkus committed:
Merge pull request #2123 from reitermarkus/spec-unlinkapps

Convert `brew unlinkapps` test to spec.

reitermarkus committed:
Merge pull request #2164 from reitermarkus/spec-bundle

Convert `brew bundle` test to spec.

reitermarkus committed:
Merge pull request #2131 from reitermarkus/spec-help

Convert `brew help` test to spec.

reitermarkus committed:
Merge pull request #2146 from reitermarkus/spec-prune

Convert `brew prune` test to spec.

reitermarkus committed:
Merge pull request #2122 from reitermarkus/spec-uninstall

Convert `brew uninstall` test to spec.

reitermarkus committed:
Merge pull request #2128 from reitermarkus/spec-unpack

Convert `brew unpack` test to spec.

reitermarkus committed:
Merge pull request #2125 from reitermarkus/spec-unlink

Convert `brew unlink` test to spec.

reitermarkus committed:
Merge pull request #2127 from reitermarkus/spec-uses

Convert `brew uses` test to spec.

reitermarkus committed:
Merge pull request #2115 from reitermarkus/spec-linkapps

Convert `brew linkapps` test to spec.

reitermarkus committed:
Merge pull request #2144 from reitermarkus/spec-readall

Convert `brew readall` test to spec.

reitermarkus committed:
Merge pull request #2124 from reitermarkus/spec-upgrade

Convert `brew upgrade` test to spec.

reitermarkus committed:
Merge pull request #2120 from reitermarkus/spec-switch

Convert `brew switch` test to spec.

reitermarkus committed:
Merge pull request #2114 from reitermarkus/spec-info

Convert `brew info` test to spec.

reitermarkus committed:
Merge pull request #2095 from joshliu/spec-deps

Convert deps test to spec

reitermarkus committed:
Avoid warnings when HOMEBREW_PREFIX/bin is not in PATH.

reitermarkus committed:
Convert ENV test to spec.

reitermarkus committed:
Convert `os/mac/diagnostic` test to spec.

reitermarkus committed:
Merge pull request #2126 from reitermarkus/spec-install

Convert `brew install` test to spec.

reitermarkus committed:
Convert `os/mac/dependency_collector` test to spec.

reitermarkus committed:
Convert `patching` test to spec.

ilovezfs committed:
audit: whitelist more unstable versions already in core

These were imported from homebrew/games.

Tenzer committed:
Add 'B' for bytes to cleanup command output

When `brew cleanup` is run it prints out a message like the following:

> This operation has freed approximately 222M of disk space.

The '222M' refers to megabytes but the normal acronym for megabytes would be
'MB'. The 'B' is also missing from kilobytes and gigabytes in the output, so
that's what this commit adds.

reitermarkus committed:
Convert Tab test to spec.

reitermarkus committed:
Add `formula` spec helper.

reitermarkus committed:
Merge pull request #2181 from reitermarkus/spec-formula-helper

Add `formula` spec helper.

reitermarkus committed:
Merge pull request #2094 from reitermarkus/spec-formula

Convert Formula test to spec.

MikeMcQuaid committed:
Merge pull request #2107 from MikeMcQuaid/audit-version-aliases

audit: check for version aliases.

MikeMcQuaid committed:
Merge pull request #2063 from MikeMcQuaid/caveats-clarify

formula: clarify caveats usage.

MikeMcQuaid committed:
Merge pull request #2093 from MikeMcQuaid/detect-recursive-dependencies

formula_installer: detect recursive dependencies.

MikeMcQuaid committed:
Merge pull request #2182 from reitermarkus/spec-tab

Convert Tab test to spec.

MikeMcQuaid committed:
Merge pull request #2178 from reitermarkus/spec-patching

Convert `patching` test to spec.

MikeMcQuaid committed:
README: sort, update maintainer list.

MikeMcQuaid committed:
README: add FX Coudert.

MikeMcQuaid committed:
Merge pull request #2170 from woodruffw/narcissism

Add myself, JCount to maintainers lists

MikeMcQuaid committed:
Merge pull request #2177 from reitermarkus/spec-os_mac_dependency_collector

Convert `os/mac/dependency_collector` test to spec.

MikeMcQuaid committed:
Merge pull request #2176 from reitermarkus/spec-os_mac_diagnostic

Convert `os/mac/diagnostic` test to spec.

MikeMcQuaid committed:
Merge pull request #2175 from reitermarkus/spec-env

Convert ENV test to spec.

vladshablinsky committed:
ARGV.include? should return boolean

reitermarkus committed:
Convert Keg test to spec.

reitermarkus committed:
Convert Options test to spec.

reitermarkus committed:
Convert `cmd/info` test to spec.

reitermarkus committed:
Convert Caveats test to spec.

reitermarkus committed:
Convert BuildOptions test to spec.

vladshablinsky committed:
system_config: check if devtools installed

Fixes #2192

EricFromCanada committed:
Work around man page generator bug for pull.rb

To work around ronn's [issue with nested
lists](https://github.com/rtomayko/ronn/issues/35), treat each item as
a separate paragraph with alternate list markers.

EricFromCanada committed:
Update brew's man page formatting and grammar

Also update command specifications to match descriptions.

EricFromCanada committed:
Update brew cask's man page formatting and grammar

Includes replacing some non-ASCII characters, which weren't being
processed properly.

reitermarkus committed:
Merge pull request #2190 from reitermarkus/spec-build_options

Convert BuildOptions test to spec.

reitermarkus committed:
Merge pull request #2188 from reitermarkus/spec-info

Convert `cmd/info` test to spec.

reitermarkus committed:
Merge pull request #2189 from reitermarkus/spec-caveats

Convert Caveats test to spec.

reitermarkus committed:
Merge pull request #2187 from reitermarkus/spec-option

Convert Options test to spec.

reitermarkus committed:
Merge pull request #2185 from reitermarkus/spec-keg

Convert Keg test to spec.

reitermarkus committed:
Merge pull request #2028 from reitermarkus/spec-migrator

Convert Migrator test to spec.

mistydemeo committed:
emoji tests: make sure to unset install badge env vars

Otherwise the user's specified emoji will be used and the test will
fail.

mistydemeo committed:
xcodebuild audit: match xcodebuild with no args

Closes #2199.

Signed-off-by: Misty De Meo <mistydemeo@gmail.com>

mistydemeo committed:
audit: fix audit on formulae without homepages

MikeMcQuaid committed:
Fix maintainers list.

Closes #2186.

MikeMcQuaid committed:
Merge pull request #2194 from EricFromCanada/master

Fix grammar, formatting in man pages

MikeMcQuaid committed:
Merge pull request #2193 from vladshablinsky/fix_tools_missing

system_config: check if devtools installed

reitermarkus committed:
Convert `cmd/update-report` test to spec.

MikeMcQuaid committed:
Merge pull request #2180 from Tenzer/cleanup-better-unit-labels

Add 'B' for bytes to cleanup command output

MikeMcQuaid committed:
Merge pull request #2201 from reitermarkus/spec-update-report

Convert `cmd/update-report` test to spec.

MikeMcQuaid committed:
Merge pull request #2112 from sjackman/needs-c++14

Add needs :cxx14

MikeMcQuaid committed:
Remove unused utils/homebrew_vars.

MikeMcQuaid committed:
bin/brew: tweak variable names.

MikeMcQuaid committed:
Merge branch 'master' into filter_all_env_vars_932

mistydemeo committed:
Add a space after the emoji for terminals

Closes #2198.

Signed-off-by: Misty De Meo <mistydemeo@gmail.com>

mistydemeo committed:
brew tests: filter out vendor/bundle

naoty committed:
Merge branch 'master' into exclude-executable-metafiles

Added tests in keg_test and pathname_test was
moved into keg_spec and pathname_spec.

MikeMcQuaid committed:
Merge pull request #2106 from naoty/exclude-executable-metafiles

Exclude executables from metafiles

MikeMcQuaid committed:
Merge pull request #1753 from doughazell/filter_all_env_vars_932

Suggested solution for 'Filter all environment variables by default #932' issue

MikeMcQuaid committed:
Merge pull request #1708 from joshka/feature/install-tap-cmd-completions

install tap cmd completions

reitermarkus committed:
Convert Cleaner test to spec.

reitermarkus committed:
Convert Dependency test to spec.

reitermarkus committed:
Convert DependencyCollector test to spec.

reitermarkus committed:
Merge pull request #2205 from reitermarkus/spec-dependency

Convert Dependency test to spec.

reitermarkus committed:
Merge pull request #2204 from reitermarkus/spec-cleaner

Convert Cleaner test to spec.

reitermarkus committed:
Merge pull request #2207 from reitermarkus/spec-dependency_collector

Convert DependencyCollector test to spec.

reitermarkus committed:
Convert FormulaInstaller test to spec.

reitermarkus committed:
Convert Dependencies and Requirements tests to specs.

MikeMcQuaid committed:
Revert "Revert "cc: always filter flags on deps.""

reitermarkus committed:
Convert formula_spec_selection test to spec.

reitermarkus committed:
Convert `brew commands` test to spec.

reitermarkus committed:
Convert Version test to spec.

reitermarkus committed:
Use `around` hook for shared "integration test" context.

reitermarkus committed:
Merge pull request #2212 from reitermarkus/spec-formula_spec_selection

Convert formula_spec_selection test to spec.

reitermarkus committed:
Merge pull request #2213 from reitermarkus/spec-commands

Convert `brew commands` test to spec.

reitermarkus committed:
Merge pull request #2209 from reitermarkus/spec-formula_installer

Convert FormulaInstaller test to spec.

reitermarkus committed:
Merge pull request #2208 from reitermarkus/spec-uninstall

Convert `brew uninstall` test to spec.

reitermarkus committed:
Merge pull request #2210 from reitermarkus/spec-dependencies

Convert Dependencies and Requirements tests to specs.

reitermarkus committed:
Convert Homebrew::Cleanup test to spec.

reitermarkus committed:
Convert Sandbox test to spec.

reitermarkus committed:
Merge pull request #2214 from reitermarkus/spec-versions

Convert Version test to spec.

reitermarkus committed:
Merge pull request #2215 from reitermarkus/spec-integration-test-helper

Use `around` hook for shared "integration test" context.

reitermarkus committed:
Convert X11Requirement test to spec.

reitermarkus committed:
Convert `formula_support` test to spec.

reitermarkus committed:
Convert `formula_validation` test to spec.

reitermarkus committed:
Convert Bottle::Filename test to spec.

reitermarkus committed:
Convert `checksum_verification` test to spec.

reitermarkus committed:
Merge pull request #2216 from reitermarkus/spec-cleanup

Convert Homebrew::Cleanup test to spec.

MikeMcQuaid committed:
Merge pull request #2211 from Homebrew/revert-1796-revert-1746-always-filter-flags-deps

Revert "Revert "cc: always filter flags on deps.""

reitermarkus committed:
Merge pull request #2217 from reitermarkus/spec-sandbox

Convert Sandbox test to spec.

reitermarkus committed:
Merge pull request #2218 from reitermarkus/spec-x11_requirement

Convert X11Requirement test to spec.

reitermarkus committed:
Merge pull request #2219 from reitermarkus/spec-formula_support

Convert `formula_support` test to spec.

reitermarkus committed:
Merge pull request #2221 from reitermarkus/spec-formula_validation

Convert `formula_validation` test to spec.

reitermarkus committed:
Merge pull request #2222 from reitermarkus/spec-bottle_filename

Convert Bottle::Filename test to spec.

reitermarkus committed:
Merge pull request #2224 from reitermarkus/spec-checksum_verification

Convert `checksum_verification` test to spec.

reitermarkus committed:
Use shorter `before` syntax to skip tests.

reitermarkus committed:
Convert Tap test to spec.

reitermarkus committed:
Convert `formula_installer_bottle` test to spec.

reitermarkus committed:
Convert FormulaText test to spec.

vladshablinsky committed:
Merge pull request #2183 from vladshablinsky/questions_mean_bool

ARGV.include? should return boolean

mistydemeo committed:
test: add a bundle config

This smooths over the transition for users who have an existing
bundle config in this location due to having gems installed at the old
path.

Closes #2203.

Signed-off-by: Misty De Meo <mistydemeo@gmail.com>

reitermarkus committed:
Merge pull request #2225 from reitermarkus/spec-before_shorthand

Use shorter `before` syntax to skip tests.

reitermarkus committed:
Merge pull request #2228 from reitermarkus/spec-audit

Convert FormulaText test to spec.

reitermarkus committed:
Merge pull request #2227 from reitermarkus/spec-formula_installer_bottle

Convert `formula_installer_bottle` test to spec.

reitermarkus committed:
Merge pull request #2226 from reitermarkus/spec-tap1

Convert Tap test to spec.

reitermarkus committed:
Convert FormulaAuditor test to spec.

reitermarkus committed:
Merge pull request #2230 from reitermarkus/spec-audit-cmd

Convert FormulaAuditor test to spec.

reitermarkus committed:
Remove Minitest.

MikeMcQuaid committed:
README: point new contributors to strict audit.

reitermarkus committed:
Add `mktmpdir` helper method.

reitermarkus committed:
Merge pull request #2232 from reitermarkus/spec-mktmpdir

Add `mktmpdir` helper method.

reitermarkus committed:
Merge Plist spec with SystemCommand::Result spec.

reitermarkus committed:
Move Formatter and Locale specs.

tchajed committed:
Fix typo in the new issue template

MikeMcQuaid committed:
Merge pull request #2239 from tchajed/fix-issue-template-typo

Fix typo in the new issue template

MikeMcQuaid committed:
README: add neutric to maintainers list.

ilovezfs committed:
bump-formula-pr: check for version presence.

If it's not there, error out rather than fetching to avoid creating
versionless cache files, which cause `brew cleanup` to crash.

reitermarkus committed:
Merge pull request #2233 from reitermarkus/remove-minitest

Remove Minitest.

ilovezfs committed:
Merge pull request #2241 from ilovezfs/bump-formula-pr-die-without-version

bump-formula-pr: check for version presence.

reitermarkus committed:
Update `.bundle/config`.

reitermarkus committed:
Merge pull request #2243 from reitermarkus/bundle-config

Update `.bundle/config`.

jonchang committed:
java_requirement: don't consider macOS Java stub

jonchang committed:
java_requirement: avoid nil in possible_javas

jonchang committed:
audit: reject versions starting with HEAD

Many parts of Homebrew assume that a version string beginning with
"HEAD" is, in fact, a head build. A stable version that begins with
"HEAD" violates this assumption and causes problems, as it's treated
as a head build in some places and as a stable build in others.

leonklingele committed:
README: Switch to https:// links

MikeMcQuaid committed:
Merge pull request #2246 from leonklingele/patch-1

README: Switch to https:// links

MikeMcQuaid committed:
Merge pull request #2245 from jonchang/head-audit

audit: reject versions starting with HEAD

fxcoudert committed:
Don't list duplicate dylibs for universal

fxcoudert committed:
Merge pull request #2251 from fxcoudert/universal

Don't list duplicate dylibs for universal

reitermarkus committed:
Merge pull request #2235 from reitermarkus/spec-plist

Merge Plist spec with SystemCommand::Result spec.

reitermarkus committed:
Merge pull request #2236 from reitermarkus/spec-formatter-locale

Move Formatter and Locale specs.

reitermarkus committed:
Use `Gem.bindir` instead of `#{Gem.dir}/bin`.

reitermarkus committed:
Simplify Cask `spec_helper`.

reitermarkus committed:
Make Cask tests work without tapping `caskroom/cask`.

reitermarkus committed:
Move Cask `shared_examples` to `test/support`.

reitermarkus committed:
Merge pull request #2256 from reitermarkus/spec-cask

Make Cask tests work without tapping `caskroom/cask`.

reitermarkus committed:
Move `shared_examples` and `matcher` into specs.

reitermarkus committed:
Remove temporary files in `uninstall` and `zap` specs.

reitermarkus committed:
Update Cask `spec_helper`.

reitermarkus committed:
Merge pull request #2257 from reitermarkus/spec-shared-examples

Move Cask `shared_examples` to `test/support`.

reitermarkus committed:
Merge pull request #2258 from reitermarkus/cask-spec-helper

Update Cask `spec_helper`.

reitermarkus committed:
Move Cask test helpers to `test/support`.

reitermarkus committed:
Move Cask specs into `brew tests`.

reitermarkus committed:
Move coverage reporting into `brew tests`.

ilovezfs committed:
audit: whitelist node@* to use conflicts_with

Add exceptions for node@* versioned formulae to use conflicts_with
instead of keg_only :versioned_formula since they, and the main node
formula, all currently overwrite npm during postinstall.

MikeMcQuaid committed:
mac/java_requirement: avoid adding/removing stub.

MikeMcQuaid committed:
Don't check Subversion HTTPS pre-Sierra.

The system Subversion doesn't handle new certificate authorities (e.g.
Let's Encrypt) well enough for this check to be useful.

MikeMcQuaid committed:
Merge pull request #2262 from MikeMcQuaid/subversion-https-audit

Don't check Subversion HTTPS pre-Sierra.

MikeMcQuaid committed:
Merge pull request #2244 from jonchang/java

java_requirement: don't consider macOS Java stub

MikeMcQuaid committed:
docs.brew.sh: use brew.sh theming.

Unfortunately there's no easy way to share this code with the brew.sh
site but it's at least able to have the `css/`, `img/` (root) and
`_layouts` directories in sync between the two repositories.

Some small CSS tweaks were needed to make these pages render more
attractively.

reitermarkus committed:
Merge pull request #2253 from reitermarkus/gem-bindir

Use `Gem.bindir` instead of `#{Gem.dir}/bin`.

reitermarkus committed:
Silence `cmd/cask` test.

reitermarkus committed:
Merge pull request #2263 from reitermarkus/spec-cmd-cask

Silence `cmd/cask` test.

reitermarkus committed:
Remove Cask Gemfile.

reitermarkus committed:
Remove `dev-cmd/tests.rb` from test coverage.

reitermarkus committed:
Use `described_class` in `Tap` spec.

reitermarkus committed:
Merge pull request #2265 from reitermarkus/spec-tap

Use `described_class` in `Tap` spec.

reitermarkus committed:
Merge pull request #2264 from reitermarkus/dev-cmd-tests

Remove `dev-cmd/tests.rb` from test coverage.

reitermarkus committed:
Add `:cask` to all Cask specs.

JCount committed:
audit: whitelist bash-completion@* to use conflicts_with

this is necessary because they install conflicting scripts into
HOMEBREW_PREFIX/etc/profile.d

reitermarkus committed:
Merge pull request #2267 from reitermarkus/move-cask-spec-helpers

Move Cask test helpers to `test/support`.

reitermarkus committed:
Merge pull request #2268 from reitermarkus/cask-spec-metadata

Add `:cask` to all Cask specs.

JCount committed:
audit: deconstruct versioned_conflicts_whitelist using splat

JCount committed:
audit: freeze versioned_conflicts_whitelist

reitermarkus committed:
Try fixing coverage.

reitermarkus committed:
Merge pull request #2270 from reitermarkus/rubocop-force-default-config

Use `--force-default-config` option for `brew cask style`.

reitermarkus committed:
Merge pull request #2260 from reitermarkus/spec-cask

Move Cask specs into `brew tests`.

reitermarkus committed:
Remove `BUNDLE_PATH` from `brew tests`.

reitermarkus committed:
Merge pull request #2271 from reitermarkus/brew-tests

Remove `BUNDLE_PATH` from `brew tests`.

EricFromCanada committed:
Add more flags and descriptions to man pages.

reitermarkus committed:
Add missing `:cask` metadata to specs.

reitermarkus committed:
Merge pull request #2273 from reitermarkus/add-missing-cask-context

Add missing `:cask` metadata to specs.

MikeMcQuaid committed:
Merge pull request #2266 from MikeMcQuaid/docs-homebrew-layout

docs.brew.sh: use brew.sh theming.

reitermarkus committed:
Move `cask` to `cmd/`.

MikeMcQuaid committed:
New-Maintainer-Checklist: add analytics link.

johnhawkinson committed:
downloads_strategy: docstring typo Gtthub -> Github

reitermarkus committed:
Use Homebrew’s `require?`.

reitermarkus committed:
Merge pull request #2275 from reitermarkus/cmd-cask

Move `cask` to `cmd/`.

JCount committed:
Merge pull request #2269 from JCount/audit-conflicts-whitelist-bash-completion

audit: whitelist bash-completion@* to use conflicts_with

reitermarkus committed:
Refactor `brew cask _audit_modified_casks`.

MikeMcQuaid committed:
Merge pull request #2272 from EricFromCanada/master

Add more flags and descriptions to man pages.

MikeMcQuaid committed:
Merge pull request #2276 from johnhawkinson/dstypo

download_strategy: docstring typo Gtthub -> Github

sjackman committed:
bottle: Remove superfluous odie

reitermarkus committed:
Make `timestamped_versions` return an array of strings.

reitermarkus committed:
Merge pull request #2277 from reitermarkus/require

Use Homebrew’s `require?`.

reitermarkus committed:
Merge pull request #2280 from sjackman/bottle-mismatches

bottle: Remove superfluous odie

reitermarkus committed:
Merge pull request #2278 from reitermarkus/cask-_audit_modified_casks

Refactor `brew cask _audit_modified_casks`.

reitermarkus committed:
Merge pull request #2282 from reitermarkus/cask-versions-string

Make `timestamped_versions` return an array of strings.

bw-yuhogg committed:
Merge tag Homebrew/1.1.11 into Linuxbrew/master

bw-yuhogg commented on Library/Homebrew/cmd/cask.rb:
```diff
@@ -0,0 +1,10 @@
+  module_function
+
+  def cask
+    Hbc::CLI.process(ARGV)
```
@Linuxbrew/core we might want to put some sort of bail-out code here for Linux.

bw-yuhogg commented on Library/Homebrew/extend/os/mac/diagnostic.rb:
```diff
@@ -53,8 +53,11 @@ def check_xcode_up_to_date
+        # `brew test-bot` runs `brew doctor` in the CI for the Homebrew/brew
+        # repository. This only needs to support whatever CI provider
+        # Homebrew/brew is currently using.
+        return if ENV["TRAVIS"]
```
FIXME: need to investigate this one and figure out why Circle is no longer mentioned.

maxim-belkin replied:
Function name is `check_xcode_up_to_date`, so I would add ```return unless OS.mac?``` in the very beginning :)

Anyways, related commit is:

```patch
$ git show 6ac0de1b
commit 6ac0de1b40699042ea124813f2e953285efbd05f
Author: Mike McQuaid <mike@mikemcquaid.com>
Date:   Fri Feb 17 21:27:40 2017 +0000

    Revert "Skip Xcode outdated check on CircleCI"

diff --git a/Library/Homebrew/extend/os/mac/diagnostic.rb b/Library/Homebrew/extend/os/mac/diagnostic.rb
index 3e0c1d0..6d30cd2 100644
--- a/Library/Homebrew/extend/os/mac/diagnostic.rb
+++ b/Library/Homebrew/extend/os/mac/diagnostic.rb
@@ -53,8 +53,8 @@ module Homebrew
         return unless MacOS::Xcode.installed?
         return unless MacOS::Xcode.outdated?
 
-        # CI images are going to end up outdated so don't complain.
-        return if ENV["TRAVIS"] || ENV["CIRCLECI"]
+        # Travis CI images are going to end up outdated so don't complain.
+        return if ENV["TRAVIS"]
```

bw-yuhogg replied:
Ah, I didn't even realize that this had nothing to do with Linuxbrew. I just saw "CIRCLECI" and assumed one of us put it there. But it's actually in extend/os/mac, so there's no action required on our end.

sjackman replied:
Note that Linuxbrew currently does import `extend/os/mac` because we still use some functions in there. That definitely needs to be fixed. Linuxbrew should not import  `extend/os/mac`.

bw-yuhogg replied:
😱

maxim-belkin replied:
🍠

sjackman replied:
The horror! The horror!

bw-yuhogg commented on Library/Homebrew/keg.rb:
```diff
@@ -207,8 +207,8 @@ def ==(other)
-    Pathname.glob("#{path}/**/*") do |file|
-      next if file.directory?
+    Pathname.glob("#{path}/*") do |file|
+      return false if file.directory? && !file.children.reject(&:ds_store?).empty?
```
FIXME: not sure what's up with this one, have to look more closely.

sjackman replied:
Looks to me that it no longer recurses into subdirectories (`*` instead of `**/*`) and it's considered non-empty if it contains a directory that its not itself empty (while ignoring Mac special files named `.DS_Store`).

bw-yuhogg committed:
formula_spec: Use xorg_requirement on Linuxbrew

Basically Shaun's commit dc911dff5d3fe2db04fd567971776c52951a5563
for formula_test, but for RSpec instead.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

maxim-belkin commented on Library/Homebrew/test/formula_spec.rb:
```diff
@@ -727,7 +727,7 @@ def test
 
     python = PythonRequirement.new
-    x11 = X11Requirement.new("x11", [:recommended])
+    x11 = (OS.mac? ? X11Requirement : XorgRequirement).new("x11", [:recommended])
```
should we move that `if OS.mac?` to `X11Requirement` which would require `XorgRequirement` on Linux platforms? This is a general question so it should not block this PR

bw-yuhogg replied:
Glad you asked. In the future, I want to rewrite the X* requirements to use the OS extension system Mike introduced upstream, in the same manner as my work on the Java and FUSE requirements.

Presumably, this would imply:

1. Moving the XQuartz logic to extend/os/mac
2. Moving the linuxbrew/xorg/xorg handling stuff to extend/os/linux
3.  (Only on Linuxbrew, not upstream) adding XorgRequirement to the compat file to point to X11Requirement

What do you think of that plan?

maxim-belkin replied:
LGTM! As long as this work makes Linuxbrew and Homebrew codebase closer together, I'm 👍 on that

bw-yuhogg committed:
cleaner_spec: Fix test for Linuxbrew

Basically Shaun's commit a8a868531649a4a83ed0cd3dc5c033cd5ba4a4ce,
but for RSpec.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
dependency_collector_spec: Partially fix for Linuxbrew

Some of the tests are only failing because :x11 maps to
XorgRequirement instead of X11Requirement on non-Mac.

Two of them also fail for other reasons and require
more investigation.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
diagnostic_spec: Needs macOS

There's not much point, IMO, in testing that the return value of
check_homebrew_prefix is nil. So just skip the whole test.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
dependecy_collector_spec: Disable two tests for Linuxbrew

XorgRequirement does not implement min_version, so these tests
are only meaningful or useful on macOS.

Perhaps it would make sense to implement that method in the
future, but since there will likely be substantial changes
to this upstream in the future there's no real reason to
do it now.

bw-yuhogg committed:
diagnostic_spec: Disable check_dyld_vars for Linuxbrew

This test considers an environment variable which only
affects macOS.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
test_spec: Disable test for Linuxbrew

Only part of the test is failing, and it has been for some time.
Equivalent to Shaun's commit cd4897a7fcdfae4875e1f662ce91fc18fa276fb2,
but for RSpec.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
uses_spec: Fix test for Linuxbrew on Travis

When HOMEBREW_VERBOSE_USING_DOTS is set, as it is on Travis,
brew uses will produce output to stderr even when there's
nothing of interest there.

To make the tests pass, force this environment variable on
and then change the tests to expect this output.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
Regenerate man pages

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
install_spec: Disable HEAD install test for Linuxbrew

This test currently fails on upstream's 1.1.11 too,
so disable it rather than have it fail incessantly.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
cask: Only enable on macOS

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
Add Linux-specific testball

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
install_spec: Disable some tests for Linuxbrew

These are failing, but more due to technicalities (SHA-256 mismatches
after applying a patch) than anything else. They're not worth
holding up the release of 1.1.11 any longer.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg committed:
README.md: Add jonchang to maintainer list

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

bw-yuhogg commented:
@Linuxbrew/core all in favour of merging this to master (but not creating a new tag)?

It'd be easier for us to catch any remaining problems if we're all using it (hint hint...)

sjackman commented:
:shipit: Go ahead and merge it to master if you think it's ready. Thanks for tackling this merge, Bob. Sounds like it was a tricky one. We can tag it as stable once we've had a couple of days to test it.

maxim-belkin commented:
I've played around with this one and further fixes require taking care of "... GCC requested but formula gcc not installed...", that is, it will require @sjackman :)

sjackman commented:
> GCC requested but formula gcc not installed...

What commands are needed to reproduce that error?

sjackman commented:
Merged, but not yet tagged.

sjackman commented:
Thanks for all of this work, Bob!

maxim-belkin commented:
> What commands are needed to reproduce that error?

```
brew tests
```

sjackman commented:
I'm not seeing that failure with `brew tests`. In which environment, or was the test disabled?
